Update kubeadm for Windows (#19217)
kubernetes/website: content/en/docs/setup/production-environment/windows/user-guide-windows-nodes.md

{{% capture prerequisites %}}

* Obtain a [Windows Server 2019 license](https://www.microsoft.com/en-us/cloud-platform/windows-server-pricing)
(or higher) in order to configure the Windows node that hosts Windows containers.
If you are using VXLAN/Overlay networking you must have also have [KB4489899](https://support.microsoft.com/help/4489899) installed.

* Build a Linux-based Kubernetes cluster in which you have access to the control-plane (some examples include [Creating a single control-plane cluster with kubeadm](/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/), [AKS Engine](/docs/setup/production-environment/turnkey/azure/), [GCE](/docs/setup/production-environment/turnkey/gce/), [AWS](/docs/setup/production-environment/turnkey/aws/).

> 

Reviewer: Is there a minimum version of Kubernetes that you need to be using? Consider adding a `{{< version-check >}}` shortcode.

>     {{< note >}}The VNI must be set to 4096 and port 4789 for Flannel on Linux to interoperate with Flannel on Windows. Support for other VNIs is coming soon. See the [VXLAN documentation](https://github.com/coreos/flannel/blob/master/Documentation/backends.md#vxlan)

Reviewer: I'd also change to:
```suggestion
    {{< note >}}The VNI must be set to 4096 and port 4789 for Flannel on Linux to interoperate with Flannel on Windows. See the [VXLAN documentation](https://github.com/coreos/flannel/blob/master/Documentation/backends.md#vxlan).
```
based on https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-about-the-future

    kubectl apply -f https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/flannel-overlay.yml
    ```

    {{< note >}}
    If you're using host-gateway use https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/flannel-host-gw.yml instead
    {{< /note >}}

### Joining a Windows worker node
{{< note >}}
You must install the `Containers` feature and install Docker. Instructions
to do so are available at [Install Docker Engine - Enterprise on Windows Servers](https://docs.docker.com/ee/docker-ee/windows/docker-ee/#install-docker-engine---enterprise).
{{< /note >}}

{{< note >}}
All code snippets in Windows sections are to be run in a PowerShell environment
with elevated permissions (Administrator) on the Windows worker node.
{{< /note >}}

1. Install wins, kubelet, and kubeadm.

   ```PowerShell
   curl.exe -LO https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/PrepareNode.ps1
   .\PrepareNode.ps1 -KubernetesVersion {{< param "fullversion" >}}
   ```

> 1. Run kubeadm to join the node

Reviewer: nit:
```suggestion
1. Run `kubeadm` to join the node
```

    After a few minutes, you should see all the pods as running if the Flannel pod network was deployed.

    ```bash
    kubectl get pods -n kube-system
    ```

    The output should include the Linux flannel DaemonSet as running:

    ```
    NAMESPACE     NAME                                      READY        STATUS    RESTARTS   AGE
    ...
    kube-system   kube-flannel-ds-54954                     1/1          Running   0          1m
    ```

1. Add Windows Flannel and kube-proxy DaemonSets

>     Now we will add Windows-compatible versions of Flannel and kube-proxy. In order

Reviewer: [Avoid using “we”](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we), if feasible.

    to ensure that we get a compatible version of kube-proxy, we'll need to substitute
    the tag of the image. The following example shows usage for Kubernetes 1.17.0,
    but you should adjust the version for your own deployment.

    ```bash

>     curl -L https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/kube-proxy.yml | sed 's/VERSION/v1.17.0/g' | kubectl apply -f -

Reviewer: ```suggestion
    curl -L https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/kube-proxy.yml | sed 's/VERSION/{{< param "fullversion" >}}/g' | kubectl apply -f -
```

    kubectl apply -f https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/flannel-overlay.yml
    ```

    {{< note >}}
    If you're using host-gateway use https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/flannel-host-gw.yml instead
    {{< /note >}}

### Joining a Windows worker node
{{< note >}}
You must install the `Containers` feature and install Docker. Instructions

> to do so can be found [here](https://docs.docker.com/ee/docker-ee/windows/docker-ee/#install-docker-engine---enterprise).

Reviewer: ```suggestion
to do so are available at [Install Docker Engine - Enterprise on Windows Servers](https://docs.docker.com/ee/docker-ee/windows/docker-ee/#install-docker-engine---enterprise).
```

{{< /note >}}

{{< note >}}
All code snippets in Windows sections are to be run in a PowerShell environment
with elevated permissions (Administrator) on the Windows worker node.
{{< /note >}}

1. Install wins, kubelet, and kubeadm.

   ```PowerShell
   curl.exe -LO https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/PrepareNode.ps1

>    .\PrepareNode.ps1 -KubernetesVersion v1.17.0

Reviewer: ```suggestion
   .\PrepareNode.ps1 -KubernetesVersion {{< param "fullversion" >}}
```

   ```
{{< note >}}

> Kubernetes version must be >= 1.17.0

Reviewer: Adding a `version-check` shortcode earlier should mean you don't need to repeat this.

{{< /note >}}

1. Run kubeadm to join the node

    Use the command that was given to you when you ran `kubeadm init` on the master node.
    If you no longer have this command, or the token has expired, you can run `kubeadm token create --print-join-command`

>     on the master to generate a new token and join command.

Reviewer: ```suggestion
    Use the command that was given to you when you ran `kubeadm init` on a control plane host.
    If you no longer have this command, or the token has expired, you can run `kubeadm token create --print-join-command`
    (on a control plane host) to generate a new token and join command.
```

#### Verifying your installation
You should now be able to view the Windows node in your cluster by running:

```bash
kubectl get nodes -o wide
```

If your new node is in the `NotReady` state it is likely because the flannel image is still downloading.
You can check the progress as before by checking on the flannel pods in the `kube-system` namespace:

> ```

Reviewer: nit: Mark this as shell or PowerShell.

kubectl -n kube-system get pods -l app=flannel
```

> Once the flannel pod is running your node should enter the `Ready` state and is ready to handle workloads.

Reviewer: nit:
```suggestion
Once the flannel Pod is running, your node should enter the `Ready` state and then be available to handle workloads.
```

{{% /capture %}}

> {{% capture next %}}

Reviewer: Did you mean:
```suggestion
{{% capture whatsnext %}}
```
?
